machine/machine_core/ssh_connection.py

```diff
@@ -437,17 +437,21 @@ def download_dir(self, source, dest, relative_dir="."):
         except:
             self.message("Error while downloading directory '{0}'".format(source))
 
-    def write(self, dest, content):
+    def write(self, dest, content, append=False, mode=None):
```

Comment:
`chmod` might be a better name than `mode`.  it's less accurate, but it's much clearer.  `mode` is too overloaded.  `perm` is another possibility.  `mode` is also fine.  your choice :)

`owner=` (or similarly, `chown=`) string e.g. `admin:admin` or so, might be a nice addition.

Reply:
Right, perm= makes sense, and I added owner= as well. We don't use it anywhere right now, but at least it's there when we want to.